scripts/tdr/copy_from_tdr_to_gcs_hca/copy_from_tdr_to_gcs_hca.py

```diff
@@ -0,0 +1,216 @@
+# Description: This script copies data from Terra Data Repository (TDR) HCA project buckets to HCA staging area buckets.
+# It is based on the bash script get_snapshot_files_and_transfer.sh, written by Samantha Velasquez.
+
+import os
+import sys
+import csv
+import logging
+import re
+import requests
+import google.auth
+import google.auth.transport.requests
+import subprocess
+
+STAGING_AREA_BUCKETS = {
+        "EBI": "gs://broad-dsp-monster-hca-prod-ebi-storage/prod",
+        "UCSC": "gs://broad-dsp-monster-hca-prod-ebi-storage/prod",
+        "LANTERN": "gs://broad-dsp-monster-hca-prod-lantern",
+        "LATTICE": "gs://broad-dsp-monster-hca-prod-lattice/staging",
+        "TEST": "gs://broad-dsp-monster-hca-prod-ebi-storage/broad_test_dataset"
+}
+
+
+def setup_cli_logging_format() -> None:
+    logging.basicConfig(level=logging.INFO, format='%(message)s', stream=sys.stdout)
+
+
+def validate_input(csv_path: str):
+    """
+    input should be a manifest csv of those projects that need data copied back
+    format is <institution>,<project_id>
+    """
+    if not os.path.isfile(csv_path):
+        logging.debug(f"{csv_path} not found")
+        sys.exit(1)
+
+    if not csv_path.endswith('.csv'):
+        logging.debug(f"{csv_path} is not a csv file")
+        sys.exit(1)
+
+    else:
+        return csv_path
+
+
+def find_project_id_in_str(s: str) -> str:
+    """
+    The selected function find_project_id_in_str(s: str) -> str:
+    is used to extract a valid UUID (Universally Unique Identifier) from a given string s.
+    :param s:
+    :return:
+    Attribution:
+    https://github.com/DataBiosphere/hca-ingest/blob/main/orchestration/hca_orchestration/support/matchers.py
+    """
+    uuid_matcher = re.compile('[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}', re.I)
+    project_ids = uuid_matcher.findall(s)
+
+    if len(project_ids) != 1:
+        raise Exception(f"Found more than one or zero project UUIDs in {s}")
+
+    return str(project_ids[0])
+
+
+def _sanitize_staging_gs_path(path: str) -> str:
+    return path.strip().strip("/")
+
+
+def _parse_csv(csv_path:str):
+    """
+    Parses the csv file and returns a list of staging areas
+    :param csv_path:
+    :return:
+    Attribution:
+    https://github.com/DataBiosphere/hca-ingest/blob/main/orchestration/hca_manage/manifest.py
+    """
+    tuple_list = []
+    with open(csv_path, "r") as f:
+        reader = csv.reader(f)
+        for row in reader:
+            if not row:
+                logging.debug("Empty path detected, skipping")
+                continue
+
+            assert len(row) == 2
```

Comment:
Maybe move this validation up to def validate_input()

Reply:
I could or I could combine them - one is generic - "is this a csv?" and the second is specific to this script